Code/Framework/AzCore/AzCore/Serialization/ObjectStream.cpp

```diff
@@ -1115,6 +1115,10 @@ namespace AZ
                     if (GenericClassInfo* genericClassInfo = sc.FindGenericClassInfo(cd->m_typeId))
                     {
                         element.m_id = genericClassInfo->GetSpecializedTypeId();
+                        if (genericClassInfo->GetGenericTypeId() == GetAssetClassId())
+                        {
+                            element.m_id = GetAssetClassId();
+                        }
```

Comment:
This code shouldn't swap out the `AZ::Data::Asset<AssetDataDerivedClass>` aggregate Id for the generic AssetClassId, instead it should update the check in [ObjectStream::LoadClass](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/ObjectStream.cpp#L769) to query the GenericClassInfo from the SerializeContext there and determine if the the aggregate id maps to the genericAssetClassId